web/src/components/storage/DeviceSelector.jsx

```diff
@@ -0,0 +1,245 @@
+ *
+ * @param {Object} props
+ * @param {StorageDevice} props.device
+ */
+const ItemContent = ({ device }) => {
+  const BasicInfo = () => {
+    const DeviceIcon = () => {
+      const names = {
+        raid: "storage",
+        md: "storage"
+      };
+
+      const name = names[device.type] || "hard_drive";
+
+      return <Icon name={name} />;
+    };
+
+    const DeviceSize = () => {
+      if (device.size === undefined) return null;
+
+      return <div>{deviceSize(device.size)}</div>;
+    };
+
+    return (
+      <div>
+        <DeviceIcon />
+        <DeviceSize />
+      </div>
+    );
+  };
+
+  const ExtendedInfo = () => {
+    const DeviceName = () => {
+      if (device.name === undefined) return null;
+
+      return <div>{device.name}</div>;
+    };
+
+    const DeviceType = () => {
+      let type;
+
+      switch (device.type) {
+        case "multipath": {
+          type = "Multipath";
+          break;
+        }
+        case "dasd": {
+          type = `DASD ${device.busId}`;
+          break;
+        }
+        case "md": {
+          type = `Software ${device.level.toUpperCase()}`;
+          break;
+        }
+        case "disk": {
+          type = device.sdCard ? "SD Card" : `Transport ${device.transport}`;
+        }
+      }
+
+      return <If condition={type} then={<div>{type}</div>} />;
+    };
+
+    const DeviceModel = () => {
+      if (!device.model || device.model === "") return null;
+
+      return <div>{device.model}</div>;
+    };
+
+    const MDInfo = () => {
+      if (device.type !== "md") return null;
+
+      const members = device.members.map(m => m.split("/").at(-1));
+
+      return <div>Members: {members.join(", ")}</div>;
+    };
+
+    const RAIDInfo = () => {
+      if (device.type !== "raid") return null;
+
+      const devices = device.devices.map(m => m.split("/").at(-1));
+
+      return <div>Devices: {devices.join(", ")}</div>;
+    };
+
+    const MultipathInfo = () => {
+      if (device.type !== "multipath") return null;
+
+      const wires = device.wires.map(m => m.split("/").at(-1));
+
+      return <div>Wires: {wires.join(", ")}</div>;
+    };
+
+    return (
+      <div>
+        <DeviceName />
+        <DeviceType />
+        <DeviceModel />
+        <MDInfo />
+        <RAIDInfo />
+        <MultipathInfo />
+      </div>
+    );
+  };
+
+  const ContentInfo = () => {
+    const PTable = () => {
+      if (device.partitionTable === undefined) return null;
+
+      const type = device.partitionTable.type.toUpperCase();
+      const numPartitions = device.partitionTable.partitions.length;
+
+      const text = `${type} with ${numPartitions} partitions`;
+
+      return (
+        <div>
+          <Icon name="folder" size="14" /> {text}
+        </div>
+      );
+    };
+
+    const Systems = () => {
+      if (device.systems.length === 0) return null;
+
+      const System = ({ system }) => {
+        const logo = /windows/i.test(system) ? "windows_logo" : "linux_logo";
```

Comment:
It is concise, but not the most readable line I have seen... 😅

Reply:
Tell it to eslint :-)